### PR TITLE
IN which I lose my sanity by slowly reverse engineering map patcher [DNM]

### PR DESCRIPTION
**Pull request by optimumtact**
This will be such fun, help welcome

### CODE REVIEW

**Comment by xxalpha** [comment]
:speedboat:

**Comment by optimumtact** [comment]
I think I understand this enough to begin rewriting it in python at least

**Comment by ike709** [comment]
honestly why was it written in java in the first place

:+1:

**Comment by PKPenguin321** [comment]
godspeed

File: tools/mapmerge/Source/src/MapPatcher Source/Map.java [hunk 1]
```diff
@@ -51,68 +52,90 @@ public Map(File paramFile, boolean paramBoolean)
...
+                if (this.minx > i2) this.minx = i2;//if currx < minx set
+                if (this.maxx < i2) this.maxx = i2;//if currx > maxx set
+                //Get tiles is a map of location -> tiletype
                 this.tiles.put(localLocation, this.tile_types.get(str4));
+                //Move string up
                 str1 = str1.substring(i);
+                //currx +1 (as we walk across
                 i2++;
               }
+              //New line curry +1
```
Comment (duncathan) [inline tools/mapmerge/Source/src/MapPatcher Source/Map.java]: mmm I love curry

**Comment by Chnkr** [comment]
This looks like :spaghetti:

**Comment by Rinoka** [comment]
For curiousity's sake, why's this necessary? I've been out of the loop.

**Comment by optimumtact** [comment]
@Chnkr I'm not suprised, it was decompiled from the jar, so it's not the original source code

@Rinoka 
1) We should probably have at least one person who understands a tool that's pretty crucial to the mappers workflow
2) It doesn't play well with remies dmm2tgm, so I'd like to join the two into one tool, which necessitates understanding what this one does and why it doesn't like dmm2tgm (answer, a naive parser)

**Comment by Rinoka** [comment]
Ah, okay. I'll turn a critical eye on it, see if I can't figure it out myself.

**Comment by Rinoka** [comment]
Okay, I get why it doesn't like to play nice with the dmm2tgm converter. It adds a header like so: "//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE". The merger is only prepared to deal with certain writing styles; this isn't one of them. Because it starts with //, it trips the merger's search function, and it saves the header as a turf (I think).

Unsure EXACTLY what it takes, but it won't compile properly on my machine, else I'd be able to do a runthrough.

**Comment by Chnkr** [comment]
I actually re-wrote the entire map-patcher, i'll PR it after i've added some unit tests.

**Comment by Rinoka** [comment]
I have a feeling the major reason that the map merger is fucked is the header; I don't think it was coded to be modular around headers and other things of that nature.

God, I wish this compiled cleanly on my machine, though. This'd be so much easier if I could just step through it.

**Comment by Rinoka** [comment]
Thank fucking christ.

**Comment by optimumtact** [comment]
@Rinoka  the mappatcher parser expects an entire "aab" = (/path/to/tile{blahbah}) to live on a single line

dmm2tgm puts it on seperate lines so it just cannot handle tgm files at all.

**Comment by Chnkr** [comment]
`dmm2tgm` is just weird, why would anyone ever want to enlargen the map file to something larger!?!

**Comment by optimumtact** [comment]
@Chnkr the original thought behind it was that git can merge more effectively on line by line changes, so if you split things up into their own lines, git can much more effectively merge and diff the changes.

**Comment by Chnkr** [comment]
kek

@optimumtact look, i rewrote the entire fucking thing to be less shit, should be done as soon as i'm sober enough to finish it.

**Comment by optimumtact** [comment]
cheers

**Comment by optimumtact** [comment]
@Chnkr soon?

**Comment by Chnkr** [comment]
@optimumtact soon(tm)

**Comment by Razharas** [comment]
@optimumtact 
@Chnkr

**Comment by optimumtact** [comment]
Still on this, just got delayed

**Comment by optimumtact** [comment]
don't' really have the motivation to continue on this right now.

Hopefully the comments so far remain useful to somebody

**Pull request closed without merging**